### DISCUSSION TITLE
Alternativa a DuckDNS con SSL

### DISCUSSION THREAD

**Discussion by Jones1982** [indomus/forum#586] [HUB: Home Assistant]
Buona sera,
come tutti (credo) ho installato e configurato l'add-on DuckDNS.
Configurato tutto, riesco ad accedere solo tramite il link esterno (es: https://casamia.duckdns.org:8123), sia da browser che da APP android, ma non mi fa l'accesso dall'interno della rete locale (tramite app), puntando direttamente all'ip (192.168.1.250:8123 -  con o senza la "s" all'http), mentre facendo l'accesso dal browser da errore di certificato etc..., ma ignorando l'avviso di sicurezza riesco cmq ad accedere.

Avendo già attivo il servizio ddns di No-Ip.org, in alternativa, posso "avere" un certificato valido/gratuito, sia x il percorso esterno (es: https://casamia.no-ip.org:8123), che per quello interno con l'ip, da usare con HA?

Ovviamente accedo tranquillamente da browser e app (sia dall'esterno che nella lan) se disattivo duckdns e la stringa in configuration.yaml del certificato

Grazie a chi risponderà

**Comment by maxcanna** [comment]
> posso "avere" un certificato valido/gratuito, sia x il percorso esterno (es: https://casamia.no-ip.org:8123), che per quello interno con l'ip, da usare con HA?

No.

Comunque il comportamento che hai descritto è del tutto normale. Se vuoi saperne di più studia come funzionano i certificati che stanno alla base di TLS e https.

PS
La s in https non è lì per decorazione. Prima di procedere ti consiglio di studiare un po' di concetti base o avrai difficoltà in futuro

**Reply by Jones1982** [reply]
Quindi attivare Duck con il suo certificato, a poco serve se lo vuoi usare anche da locale!
Si, ho visto nei dettagli del browser sul certificato che indicava che era valido solo x l'indirizzo creato con duck.
la S dell'http(s) si lo so a cosa si riferisce. era x specificare bene ogni cosa.

**Reply by maxcanna** [reply]
Beh usa sempre il dominio e via no? 😉

PS
Fattelo sto studio, dai retta ☺

**Reply by Jones1982** [reply]
> Beh usa sempre il dominio e via no? 

E' grazie tante...! ahahha
E il tablet attaccato al muro, quando internet va KO, come si collega al dominio (esterno) poi? 
Sarei vincolato h24 a internet. Se cade, o si blocca, o altri ca**i... non avrei accesso. 
L'ho sto usando da quasi 1 anno senza certificato cmq, e non ho problemi. Però volevo (potendo) usarlo col certificato x ovvi motivi (quando sto fuori casa etc..).
La cosa che mi farebbe comodo, è usarlo col certificato tramite l'uso di duck da fuori casa, e internamente senza...in contemporanea. ma deduco che non si possa fare...!

**Reply by maxcanna** [reply]
Ero serio. Lo vedi che devi studiare? Il fatto che il dominio sia "esterno" come lo chiami tu non implica che tu abbia bisogno di internet. Però oh, casa tua regole tue ☺

**Reply by Jones1982** [reply]
> Ero serio. Lo vedi che devi studiare? Il fatto che il dominio sia "esterno" come lo chiami tu non implica che tu abbia bisogno di internet. Però oh, casa tua regole tue

Scusa è, se l'app pretende un indirizzo per HA x il collegamento in locale (lan) e x collegamento esterno (e lo chiama lui così, non io), è ovvio che se sto SEMPRE connesso sull'indirizzo esterno, quando internet si blocca, cade o altro, il collegamento con HA si blocca (dall'app) sul tablet connesso nella mia lan!

Poi che mi rispondi, e ti ringrazio è!, ripetendo solo "STUDIA", è come rispondere a uno che cerca un benzinaio, usa il navigatore.
scusa tanto è.

T ringrazio x la risposta, ma sa molto di "aaa che palle sti principianti...".
Gradivo di più un SI, NO, NO xchè questo.  eheehehe

**Reply by maxcanna** [reply]
Non è quello.

È che io ho percepito della confusione e ti ho consigliato di studiare come trovarlo da solo il benzinaio. Così sei autonomo e magari sai trovare anche qualcos'altro se ti dovesse servire. Invece tu preferisci che io ti dica "al semaforo a destra e poi sempre dritto".

Dal mio punto di vista non è il modo di procedere quindi scusami ma per me finiamo qui ☺

**Reply by Jones1982** [reply]
Scusami tanto se al momento della registrazione, mi è sfuggito il msg: "Forum x soli esperti del settore." haaha :D

Giusto per chiudere la polemica, visto che ti sei seccato, manco ti avessi offeso...

ma risponde alla domanda con...: "se ti interessa leggiti 4000 post, studiati 10 siti specializzati e svariate guide", x poi arrivare alla conclusione che cmq devo comprarlo il certificato, tanto valeva che non mi rispondevi affatto.

Chi posta domande qui, si aspetta (e spera) di trovare una risposta/aiuto e magari una spiegazione (in grandi linee), e non un "Studia e non rompere".   ;o)) 

Cmq vabbè, non è accaduto nulla....!!! Grazie lo stesso!

**Comment by snakuzzo** [comment]
Qui le cose sono due:

- utilizzare un DDNS diverso da DuckDNS
- accedere ad HA da esterno con FQDN e dall'iterno con IP

Per avere entrambe le cose dovresti installare un reverse proxy. Su HassOS molti utilizzano l'addon [NGINX Proxy Manager](https://github.com/hassio-addons/addon-nginx-proxy-manager) che se non vado errato dovrebbe offrirti anche il rilascio automatico del certificato tramite Let's Encrypt (e quindi a quel punto usi il DDNS che ti pare...anche quello noip che hai già).
Ovviamente a quel punto sarà superfluo l'addon DuckDNS che potrai cestinare.

Con questa configurazione da HA eliminerai il blocco `http:` esponendolo localmente in HTTP, mentre il reverse proxy si occuperà di esporre la tua istanza di HA in https e di rigirare le chiamate verso HA

Una volta attivato tutto, a quel punto sull'app puoi impostare ad esempio...

```
ext url  -> https://casamia.no-ip.org:9000
int url  -> http://192.168.1.250:8123
```

Attenzione alle porte perché sullo stesso server avrai sia il RP che HA e non potranno essere entrambi in ascolto sulla stessa porta. Quindi magari lasci HA sulla 8123 ed il RP sulla 9000 (è solo un esempio) e come regola di port forwarding sul tuo modem/router ovviamente farai in modo che la 9000 forwardi verso la 8123 del tuo server.

P.S. Quanto al discorso che faceva @maxcanna io credo che abbia pienamente ragione. Siamo tutti qui, in chat, su FB, sul blog per aiutare, non per prendere in giro la gente. L'impressione che ha avuto Max l'ho avuta anche io quando ho letto il primo post. Quando ho letto "con o senza la s", mi è sembrato che tu stessi improvvisando del tutto ... e questo non è positivo.
Bisogna, purtroppo o per fortuna, studiare dei concetti base...stiamo parlando di un software opensource e non di un prodotto chiavi in mano. Se si vuole un prodotto chiavi in mano da tirar su senza avere un minimo di base è meglio lasciar perdere HA e farsi la domotica con Google Assistant o con Alexa. Molti si aspettano di essere aiutati con la "pappa pronta"...è vero, ci sono le guide passo passo, ma quello che vogliamo far capire a chi ci segue è che non basta eseguire i passi di una guida sperando che funzioni e poi "chi s'è visto s'è visto". Cerchiamo anche di dare semplicemente delle imbeccate per spronare gli utenti a muoversi con le proprie gambe...e questo, ripeto, non lo facciamo perché ci rompiamo le balle o per deridere chi ne sa di meno, ma perché per noi (parlo almeno per me e per max) è il modo giusto per far crescere gli utenti

**Reply by Jones1982** [reply]
Ciao @snakuzzo , è sempre un piacere leggere una tua risposta.

Queste info cercavo, x capire se ne valeva la pena perderci tempo, etc.. 
ho capito cosa hai descritto, e poi proverò con calma...e come sempre ti ringrazio!!!

> P.S. Quanto al discorso che faceva @maxcanna io credo che abbia pienamente ragione. Siamo tutti qui, in chat, su FB, sul blog per aiutare, non per prendere in giro la gente. L'impressione che ha avuto Max l'ho avuta anche io quando ho letto il primo post. Quando ho letto "con o senza la s", mi è sembrato che tu stessi improvvisando del tutto ... e questo non è positivo.

Si ho sbagliato nel modo di descrivere le prove che avevo fatto, semplicemente dicendo "con o senza la S", ma era per dire che avevo provato anche ad accedere senza la modalità secure al protocollo http con quel tipo di settaggi, senza successo.

> Bisogna, purtroppo o per fortuna, studiare dei concetti base...stiamo parlando di un software opensource e non di un prodotto chiavi in mano. Se si vuole un prodotto chiavi in mano da tirar su senza avere un minimo di base è meglio lasciar perdere HA e farsi la domotica con Google Assistant o con Alexa. Molti si aspettano di essere aiutati con la "pappa pronta"...è vero, ci sono le guide passo passo, ma quello che vogliamo far capire a chi ci segue è che non basta eseguire i passi di una guida sperando che funzioni e poi "chi s'è visto s'è visto". Cerchiamo anche di dare semplicemente delle imbeccate per spronare gli utenti a muoversi con le proprie gambe...e questo, ripeto, non lo facciamo perché ci rompiamo le balle o per deridere chi ne sa di meno, ma perché per noi (parlo almeno per me e per max) è il modo giusto per far crescere gli utenti

Ma certo, non ho mica chiesto di settarmi il tutto xchè lo pretendevo. Ma almeno, come dici tu, l'imbeccata... e non un "studia e stammi bene".

Però la tua risposta è nettamente differente dalla sua... più sbrigativa e scocciata, che cmq ringrazio per il tempo che ha perso.

**Reply by snakuzzo** [reply]
Ok...comunque credo tu abbia frainteso il tono di Max ;)
